src/main/java/org/jenkinsci/plugins/github_branch_source/Endpoint.java

```diff
@@ -115,14 +116,27 @@ public FormValidation doCheckApiUri(@QueryParameter String apiUri) {
                 URL api = new URL(apiUri);
                 GitHub github = GitHub.connectToEnterpriseAnonymously(api.toString());
                 github.checkApiUrlValidity();
-                return FormValidation.ok();
+                LOGGER.log(Level.FINE, "Trying to configure a GitHub Enterprise server");
+                return FormValidation.ok("GitHub Enterprise server verified");
             } catch (MalformedURLException mue) {
-                return FormValidation.error("This does not look like a GitHub Enterprise API URL");
+                // For example: https:/api.github.com
+                LOGGER.log(Level.WARNING, "Trying to configure a GitHub Enterprise server: " + apiUri);
+                return FormValidation.error("This does not look like a GitHub Enterprise API endpoint");
             } catch (JsonParseException jpe) {
-                return FormValidation.error("This does not look like a GitHub Enterprise API URL");
+                LOGGER.log(Level.WARNING, "Trying to configure a GitHub Enterprise server: " + apiUri);
+                return FormValidation.error("This does not look like a GitHub Enterprise API endpoint");
+            } catch (FileNotFoundException fnt) {
+                // For example: https://github.mycompany.com/server/api/v3/ gets a FileNotFoundException
+                LOGGER.log(Level.WARNING, "Getting HTTP Error 404 for " + apiUri);
+                return FormValidation.error("This does not look like a GitHub Enterprise API endpoint");
             } catch (IOException e) {
+                // For example: https://github.mycompany.com/api/v3/ or https://github.mycompany.com/api/v3/mypath
+                if (e.getMessage().contains("private mode enabled")) {
+                    LOGGER.log(Level.FINE, e.getMessage());
+                    return FormValidation.warning("Private mode enabled, validation disabled");
+                }
                 LOGGER.log(Level.WARNING, e.getMessage());
-                return FormValidation.error("This does not look like a GitHub Enterprise API URL");
+                return FormValidation.error("This does not look like a GitHub Enterprise API endpoint");
```

Comment:
@recena That commit addresses the log, but what about the form validation?  I feel that may actually be more important, since users are more likely to see it than the logs.